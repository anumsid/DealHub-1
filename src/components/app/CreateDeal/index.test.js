import React from 'react'
import { shallow } from 'enzyme'
import { stub } from 'sinon'
import toJson from 'enzyme-to-json'
import { CreateDealForm } from './index'

describe('Create Deal Component', () => {
  let wrapper
  let onClick
  beforeEach(() => {
    onClick = stub()
    const props = {
      status: null,
      onClick: onClick
    }
    wrapper = shallow(<CreateDealForm {...props} />)
  })

  it('Matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('Handles click on form submit', () => {
    wrapper.find('FormButton').simulate('click')
    expect(onClick.calledOnce).toBe(true)
  })
})
