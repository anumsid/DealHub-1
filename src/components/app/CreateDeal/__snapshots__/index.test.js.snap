// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Deal Component Matches snapshot 1`] = `
<div
  className="createNewDeal"
>
  <br />
  <h3>
    Add New Deal
  </h3>
  <Form
    as="form"
  >
    <Form
      as="form"
      className="mx-auto large"
      style={
        Object {
          "width": "40rem",
        }
      }
    >
      <FormInput
        as={[Function]}
        control={[Function]}
        fluid={true}
        label="title"
        name="title"
        onChange={[Function]}
        placeholder="title"
      />
      <FormInput
        as={[Function]}
        control={[Function]}
        fluid={true}
        label="location"
        name="location"
        onChange={[Function]}
        placeholder="location"
      />
      <FormTextArea
        as={[Function]}
        control={[Function]}
        fluid={true}
        label="description"
        onChange={[Function]}
        placeholder="describe this deal"
      />
      <FormInput
        as={[Function]}
        control={[Function]}
        fluid={true}
        label="expiration"
        name="expiration"
        onChange={[Function]}
        placeholder="expiration"
        type="date"
      />
      <FormSelect
        as={[Function]}
        control={[Function]}
        fluid={true}
        label="Categories"
        onChange={[Function]}
        options={
          Array [
            Object {
              "key": "food",
              "text": "Food",
              "value": "food",
            },
            Object {
              "key": "beverages",
              "text": "Beverages",
              "value": "beverages",
            },
            Object {
              "key": "groceries",
              "text": "Groceries",
              "value": "groceries",
            },
            Object {
              "key": "technology",
              "text": "Technology",
              "value": "technology",
            },
            Object {
              "key": "electronics",
              "text": "Electronics",
              "value": "electronics",
            },
            Object {
              "key": "apparel",
              "text": "Apparel",
              "value": "apparel",
            },
            Object {
              "key": "auto",
              "text": "Auto",
              "value": "auto",
            },
            Object {
              "key": "home",
              "text": "Home",
              "value": "home",
            },
            Object {
              "key": "sports & outdoors",
              "text": "Sports & Outdoors",
              "value": "sports & outdoors",
            },
            Object {
              "key": "pet supplies",
              "text": "Pet Supplies",
              "value": "pet supplies",
            },
          ]
        }
        placeholder="Select Category"
      />
      <FormInput
        as={[Function]}
        control={[Function]}
        fluid={true}
        label="image [OPTIONAL]"
        name="image"
        onChange={[Function]}
        placeholder="url"
      />
    </Form>
    <br />
    <FormButton
      as={[Function]}
      control={[Function]}
      onClick={[Function]}
    >
      Submit Deal
    </FormButton>
  </Form>
</div>
`;
